### PR TITLE
finished implementing.

### PR DESCRIPTION
**Pull request by bfpepper**
Result for the validating string challenge.

### CODE REVIEW

**Comment by jcasimir** [comment]
👍

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      json (>= 1.8, < 3)
+      simplecov-html (~> 0.10.0)
+    simplecov-html (0.10.0)
+    slop (3.6.0)
+
+PLATFORMS
+  ruby
+
+DEPENDENCIES
+  minitest-nyan-cat
```
Comment (mikedao) [inline Gemfile.lock:41]: 👍

File: lib/string_validator.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class StringValidator
+
+  def validate(string)
+    valid_brackets = []
+    brackets = { '{' => '}', '[' => ']', '(' => ')' }
+    string.each_char do |char|
```
Comment (mikedao) [inline lib/string_validator.rb:6]: Is `each_char` more efficient, or is chars-ing the string and then mapping over it more efficient?

File: test/string_validator_test.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  def test_it_can_invalidate_a_string
+    assert_equal false, string.validate("([)]")
+  end
+
+  def test_it_can_validate_a_more_complex_string
+    assert_equal true, string.validate("(thi{s is a} [t]hing)")
+  end
+
+  def test_it_invalidates_messed_up_brackets
+    assert_equal false, string.validate("([)]")
```
Comment (mikedao) [inline test/string_validator_test.rb:34]: I like the sad path testing.

File: lib/string_validator.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+      valid_brackets << char if brackets.key?(char)
+      return false if brackets.key(char) && brackets.key(char) != valid_brackets.pop
```
Comment (mikedao) [inline lib/string_validator.rb:8]: This is a good place to use a predicate method in place of the conditional.

**Review by mikedao — APPROVED** [review]

**Pull request closed without merging**